src/mlpack/methods/cf/cf.hpp

```diff
@@ -88,78 +72,90 @@ class CF
      const size_t rank = 0);
 
   /**
-   * Initialize the CF object using an instantiated factorizer, immediately
+   * Initialize the CF object using any decomposition method, immediately
    * factorizing the given data to create a model. There are parameters that can
    * be set; default values are provided for each of them. If the rank is left
    * unset (or is set to 0), a simple density-based heuristic will be used to
    * choose a rank.
    *
-   * The provided dataset should be a coordinate list; that is, a 3-row matrix
+   * The provided dataset can be a coordinate list; that is, a 3-row matrix
    * where each column corresponds to a (user, item, rating) entry in the
-   * matrix.
+   * matrix or a sparse matrix representing (user, item) table.
    *
-   * @param data Data matrix: coordinate list or dense matrix.
-   * @param factorizer Instantiated factorizer object.
+   * @param data Data matrix: dense matrix (coordinate lists) 
+   *    or sparse matrix(cleaned).
+   * @param decomposition Instantiated DecompositionPolicy object.
    * @param numUsersForSimilarity Size of the neighborhood.
    * @param rank Rank parameter for matrix factorization.
+   * @param maxIterations Maximum number of iterations.
+   * @param minResidue Residue required to terminate.
+   * @param mit Whether to terminate only when maxIterations is reached.
    */
-  template<typename FactorizerType = amf::NMFALSFactorizer>
-  CF(const arma::mat& data,
-     FactorizerType factorizer = FactorizerType(),
+  template<typename mat_type>
```

Comment:
Please use camel casing for all names; see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more information.